pkg/controllers/managementuser/rbac/prtb_handler.go

```diff
@@ -109,6 +121,16 @@ func (p *prtbLifecycle) syncPRTB(binding *v3.ProjectRoleTemplateBinding) error {
 		return errors.Wrap(err, "couldn't ensure roles")
 	}
 
+	if err := p.ensureProjectRoles(roles); err != nil {
+		if errors.Is(err, collections.ErrNotReady) {
+			logrus.Debugf("not finished setting up project roles, retrying")
+			p.m.roleTemplates.Controller().EnqueueAfter(binding.Namespace, binding.Name, enqueueAfterPeriod)
+			return nil
+		}
+
+		return errors.Wrap(err, "couldn't ensure project roles")
+	}
+
 	for _, n := range namespaces {
```

Comment:
I think you need code here to handle collection RBAC for the individual namespaces in the project too.

For example, say user `test-user` has a PRTB giving them owner on project `p-123xyz`. Namespaces `test-ns-1` and `test-ns-2` are in `test-proj`. If `test-user` uses this endpoint to filter on `projectsornamespaces=p-123xyz`, I would expect this all to work, since the PRTB handler has handled creating the collection RBAC to give `test-user` access to that endpoint. However, if the user filtered on `projectsornamespaces=test-ns-1`, I don't think that this would work, since the PRTB handler only creates RBAC for the project and the collections RBAC controllers skips namespaces which are in a project. Correct me if I am missing something though, it's possible that I'm missing a part of the RBAC.

Reply:
This should still work, because it's not the query that matters for RBAC, it's the path. So I could have /namespaces/p-123xyz/pods?fieldSelector=projectsornamespaces=test-ns-1, and as long as I have permission to get pods for project p-123xyz it should work.